libs/features/personalization/personalization.js

```diff
@@ -448,6 +449,10 @@ function modifySelectorTerm(termParam) {
     term = updateEndNumber(endNumber, term);
     return term;
   }
+  if (containerSelectors.includes(startText)) {
+    term = `main > div:nth-child(${endNumber || 'n'} of main > div:has([class*="${startText}"])`;
+    return term;
```

Comment:
In here I think you can just return the expression directly, there's no utility in caching it.
```suggestion
    return `main > div:nth-child(${endNumber || 'n'} of main > div:has([class*="${startText}"])`;
```